tests/moduletool/test_install.py

```diff
@@ -941,3 +942,372 @@ def test_install_project_with_install_mode_master(tmpdir: py.path.local, snippet
+        import {module.ModuleV2.get_name_from_metadata(modtwo)}
+        """,
+        autostd=False,
+        python_package_sources=[local_module_package_index, index.url],
+        python_requires=v2_requirements,
+        install_project=True,
+    )
+
+    expected_logs = [
+        ("Successfully installed inmanta-module-modone-3.1.2", logging.DEBUG),
+        ("Successfully installed inmanta-module-modtwo-2.2.2", logging.DEBUG),
+    ]
+
+    for message, level in expected_logs:
+        log_contains(
+            caplog,
+            "inmanta.env",
+            level,
+            message,
+        )
+
+
+@pytest.mark.slowtest
+def test_git_clone_output(snippetcompiler_clean, caplog, modules_v2_dir):
+    """
+    This test checks that git clone output is correctly logged on module install.
+    """
+    caplog.set_level(logging.DEBUG)
+
+    project = snippetcompiler_clean.setup_for_snippet(
+        """
+        import std
+        """,
+        autostd=False,
+        install_project=True,
+    )
+
+    expected_logs = [
+        ("Cloning into '%s'..." % os.path.join(project.downloadpath, "std"), logging.DEBUG),
+    ]
+
+    for message, level in expected_logs:
+        log_contains(
+            caplog,
+            "inmanta.env",
+            level,
+            message,
+        )
+
+
+@pytest.mark.slowtest
+def test_no_matching_distribution(local_module_package_index: str, snippetcompiler_clean, caplog, modules_v2_dir, tmpdir):
+    """
+    Make sure the logs contain the correct message when no matching distribution is found during install.
+    """
+    caplog.set_level(logging.DEBUG)
+
+    index: PipIndex = PipIndex(artifact_dir=os.path.join(str(tmpdir), ".custom-index"))
+
+    # Scenario 1
+    # parent_module requires child_module v3.3.3 which is not installed yet.
+
+    parent_module: module.ModuleV2Metadata = module_from_template(
+        os.path.join(modules_v2_dir, "minimalv2module"),
+        os.path.join(str(tmpdir), "parent_module"),
+        new_version=version.Version("1.2.3"),
+        new_name="parent_module",
+        install=False,
+        new_requirements=[InmantaModuleRequirement.parse("child_module==3.3.3")],
+        publish_index=index,
+    )
+
+    with pytest.raises(ModuleNotFoundException):
+        snippetcompiler_clean.setup_for_snippet(
+            f"""
+            import {module.ModuleV2.get_name_from_metadata(parent_module)}
+            """,
+            autostd=False,
+            python_package_sources=[local_module_package_index, index.url],
+            python_requires=[Requirement.parse(module.ModuleV2Source.get_package_name_for("parent_module"))],
+            install_project=True,
+        )
+    log_contains(
+        caplog,
+        "inmanta.env",
+        logging.DEBUG,
+        "No matching distribution found for inmanta-module-child-module==3.3.3",
+    )
+
+    # Scenario 2
+    # parent_module requires child_module v3.3.3 which is installed with v1.1.1
```

Comment:
```suggestion
    # parent_module requires child_module v3.3.3 but the index only has v1.1.1
```
Is this what you meant? I found the wording pretty confusing.